proto/api.go

```diff
@@ -51,41 +52,187 @@ const (
 	isWrite                // write cmds go through raft and must be proposed on leader
 	isTxnWrite             // txn write cmds start heartbeat and are marked for intent resolution
 	isRange                // range commands may span multiple keys
+	isReverse              // reverse commands traverse ranges in descending direction
 )
 
 // IsAdmin returns true if the request requires admin permissions.
 func IsAdmin(args Request) bool {
+	if br, ok := args.(*BatchRequest); ok {
```

Comment:
I don't think you have quite the same constraints but this is provisional code anyways, written in that way to keep all of the old interfaces in place. I'd rather not have those things change from under me: Let's just duplicate the logic for the time being and then marry it later. Once I don't have to support `proto.Request` on the coordinator any more, I would like to remove a lot of this.